ui/src/app/applications/components/applications-list/applications-list.tsx

```diff
@@ -582,6 +582,13 @@ export const ApplicationsList = (props: RouteComponentProps<{}>) => {
                                                                 createApp={async app => {
                                                                     setAppCreatePending(true);
                                                                     try {
+                                                                        // Namespace may be specified in the app name. We need to
+                                                                        // parse and handle it accordingly.
+                                                                        if (app.metadata.name.includes('/')) {
+                                                                            const nns = app.metadata.name.split('/', 2);
+                                                                            app.metadata.name = nns[1];
+                                                                            app.metadata.namespace = nns[0];
+                                                                        }
```

Comment:
Hm, does it maybe make more sense to move this logic to https://github.com/argoproj/argo-cd/blob/43dd2b361cd95899ff5c1de86af5ae8ff9146f9c/ui/src/app/shared/services/applications-service.ts#L122 instead?

Reply:
I'd prefer the logic to be in `application-create-panel.tsx`, just incase we need to add any validation later rather service layer. Again i've very limited knowledge on the UI side, so might be incorrect.

Reply:
@jannfis I agree, I think it makes sense to put this in `applications-service.tsx`